libs/azure-storage/README.md

```diff
@@ -1,9 +1,147 @@
 # langchain-azure-storage
 
-This package contains the LangChain integrations for Azure Storage. This project is a work-in-progress and currently does not contain any features.
+This package contains the LangChain integrations for Azure Storage. Currently, it includes:
+- `AzureBlobStorageLoader` - Loads LangChain document objects from Azure Blob Storage
 
 ## Installation
 
 ```bash
 pip install -U langchain-azure-storage
 ```
+
+## Document Loader Examples
+
+### Load from container
+Below shows how to load documents from all blobs in a given container in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+The example below shows how to load documents from blobs in a container with a given prefix:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    prefix="test",
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Load from a list of blobs
+Below shows how to load documents from a list of blobs in Azure Blob Storage:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+loader = AzureBlobStorageLoader(
+    account_url=<your_account_url>,
+    container_name=<your_container_name>,
+    blob_names=["blob-1", "blob-2", "blob-3"],
+)
+
+for doc in loader.lazy_load():
+    print(doc.page_content)
+```
+
+### Load asynchronously
+Below shows how to load documents asynchronously which can be done by using either `aload()` or `alazy_load()`:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+async def main():
+    loader = AzureBlobStorageLoader(
+        account_url=<your_account_url>,
+        container_name=<your_container_name>,
+    )
+
+    async for doc in loader.alazy_load():
+        print(doc.page_content)
+```
+
+
+### Override loader
+Below shows how to override the default loader used to parse blobs:
+
+```python
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+from langchain_unstructured import UnstructuredLoader
```

Comment:
For the example, maybe let's use the `PyPDFLoader` to give an example of loading PDFs to avoid needing to introduce unstructured? We'll also show the unstructured loader later in the migration section. We should also maybe show the iteration through documents to drive in the point that the documents are directly lazy loaded from the provided loader.